src/domain/synthetics/leaderboards/usePositionsInfo.ts

```diff
@@ -0,0 +1,276 @@
+
+    const liquidationPrice = getLiquidationPrice({
+      marketInfo,
+      collateralToken,
+      sizeInUsd: position.sizeInUsd,
+      sizeInTokens: position.sizeInTokens,
+      collateralUsd,
+      collateralAmount: position.collateralAmount,
+      userReferralInfo: undefined,
+      minCollateralUsd,
+      pendingBorrowingFeesUsd: position.pendingBorrowingFeesUsd,
+      pendingFundingFeesUsd,
+      isLong: position.isLong,
+    });
+
+    positionsMap[positionKey] = {
+      ...position,
+      marketInfo,
+      indexToken,
+      collateralToken,
+      pnlToken,
+      markPrice,
+      entryPrice,
+      liquidationPrice,
+      collateralUsd,
+      remainingCollateralUsd,
+      remainingCollateralAmount,
+      hasLowCollateral,
+      leverage,
+      pnl,
+      pnlPercentage,
+      pnlAfterFees,
+      pnlAfterFeesPercentage,
+      netValue,
+      closingFeeUsd,
+      pendingFundingFeesUsd,
+      pendingClaimableFundingFeesUsd,
+    };
+
+    return positionsMap;
+  },
+  {} as PositionsData);
+};
+
+export function usePositionsInfo(
+  chainId: number,
+  positionsHash: string,
+  positionKeys: string[],
+  marketPrices: ContractMarketPrices[],
+): PositionsResult {
+  const { minCollateralUsd } = usePositionsConstants(chainId);
+  const { marketsInfoData, tokensData, pricesUpdatedAt } = useMarketsInfo(chainId);
+  const [positions, setPositions] = useState<PositionsInfoData>();
+  const positionsData = useMulticall(chainId, "usePositionsData", {
+    key: [positionsHash, pricesUpdatedAt],
+    refreshInterval: null, // Refresh on every prices update
+    request: () => ({
+      reader: {
+        contractAddress: getContract(chainId, "SyntheticsReader"),
+        abi: SyntheticsReader.abi,
+        calls: positionKeys && positionKeys.length ? {
+          positions: {
+            methodName: "getAccountPositionInfoList",
+            params: [
+              getContract(chainId, "DataStore"),
+              getContract(chainId, "ReferralStorage"),
+              positionKeys,
+              marketPrices,
+              ethers.constants.AddressZero, // uiFeeReceiver
+            ],
+          },
+        } : {},
+      },
+    }),
+  });
+
+  const hasData = Array.isArray(positionsData?.data?.data?.reader?.positions?.returnValues);
+
+  useEffect(() => {
+    const positions = positionsData?.data?.data?.reader?.positions?.returnValues;
+    if (positions && marketsInfoData && tokensData && minCollateralUsd) {
+      const parsedData = parsePositionsInfo(
+        positionKeys,
+        positions as Array<PositionJson>,
+        marketsInfoData,
+        tokensData,
+        minCollateralUsd
+      );
+      setPositions(parsedData);
+    }
+  // The dependencies below derive from the values used within the effect callback
+  // eslint-disable-next-line
+  }, [
+    positionsHash,
+    pricesUpdatedAt,
+    hasData,
+  ]);
+
+  const isLoading = !(positions && marketsInfoData && tokensData && minCollateralUsd);
```

Comment:
Maybe it's my personal taste, but,  wouldn't the use of (||) more clear?

```
 !positions || !marketsInfoData || !tokensData ...
```

Reply:
agree it is probably a matter of taste, i think for me personally it's a bit easier to read `!positions || !marketsInfoData ...` too 🤔